internal/pkg/manifest/validate.go

```diff
@@ -875,6 +882,33 @@ func (p Percentage) Validate() error {
 	return nil
 }
 
+// Validate returns nil if ScalingConfigOrPercentage is configured correctly.
+func (r ScalingConfigOrPercentage) Validate() error {
+	if r.IsEmpty() {
+		return nil
+	}
+	if r.Value != nil {
+		return r.Value.Validate()
+	}
+	return r.ScalingConfig.Validate()
+}
+
+// Validate returns nil if AdvancedScalingConfig is configured correctly.
+func (r AdvancedScalingConfig) Validate() error {
+	if r.IsEmpty() {
+		return nil
+	}
+	if err := r.Value.Validate(); err != nil {
+		return err
+	}
+	return nil
```

Comment:
tiny nit:
```suggestion
	return r.Cooldown.Validate()
```
I think it makes sense to still make this function call here because `AdvancedScalingConfig` doesn't know whether cooldown's validate is a no-op or not.